.github/workflows/test-ui-e2e.yml

```diff
@@ -0,0 +1,28 @@
+name: UI E2E (Cypress) Tests
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'frontend/**'
+      - 'test/e2e/frontend/**'
+  pull_request:
+    branches: [ main ]
```

Comment:
Should we be only running this on PRs that match the same paths as the push?

Reply:
you're absolutely right - will update